src/lib/variants.ts

```diff
@@ -69,6 +75,39 @@ export const selectVariant = (test: Test, mvtId: number): Variant => {
     return test.variants[mvtId % test.variants.length];
 };
 
+export const getUserCohort = (targeting: EpicTargeting): UserCohort => {
+    const { showSupportMessaging, isRecurringContributor } = targeting;
+
+    const lastOneOffContributionDate = targeting.lastOneOffContributionDate
+        ? new Date(targeting.lastOneOffContributionDate)
+        : undefined;
+
+    // User is a current supporter if she has a subscription or a recurring
+    // donation or has made a one-off contribution in the past 3 months.
+    const isSupporter =
+        !showSupportMessaging ||
+        isRecurringContributor ||
+        isRecentOneOffContributor(lastOneOffContributionDate);
+
+    // User is a past-contributor if she doesn't have an active subscription
+    // or recurring donation, but has made a one-off donation longer than 3
+    // months ago.
+    const isPastContributor =
+        !isSupporter &&
+        lastOneOffContributionDate &&
+        !isRecentOneOffContributor(lastOneOffContributionDate);
+
+    if (isPastContributor) {
+        return 'PostAskPauseSingleContributors';
+    } else if (isSupporter) {
+        return 'AllExistingSupporters';
+    } else if (!isSupporter) {
+        return 'AllNonSupporters';
+    } else {
+        return 'Everyone';
```

Comment:
How can it get to the `Everyone` case here?

Reply:
Good question. You can't. The 'Everyone` cohort is only ever set in the test as a means to bypass the cohort checks altogether. So when determining the cohort a user falls into, it will always fall on one of the other 3.
I've tweaked this function to return `AllNonSupporters` instead by default.